addons/website_event_sale/controllers/sale.py

```diff
@@ -0,0 +1,16 @@
+# -*- coding: utf-8 -*-
+# Part of Odoo. See LICENSE file for full copyright and licensing details.
+
+from odoo.addons.website_sale.controllers.main import WebsiteSale
+
+
+class WebsiteEventSale(WebsiteSale):
+
+    def _prepare_confirmation_values(self, order):
+        values = super(WebsiteEventSale,
+                       self)._prepare_confirmation_values(order)
+        if any(line.product_id.detailed_type == 'event' for line in order.order_line):
```

Comment:
Note that as you do a t-if="events", and as event_id should always come wiht a product of detailed type event, I think it could be simplified with
```
values['events'] = order.order_line.event_id
```
which will be a void recordset if no events found. That way we have a one-liner.